pkg/runtime/wfengine/wfengine.go

```diff
@@ -46,37 +42,56 @@ type WorkflowEngine struct {
 	actorRuntime   actors.Actors
 	startMutex     sync.Mutex
 	disconnectChan chan any
+	config         *wfConfig
 }
 
 var (
 	wfLogger            = logger.NewLogger("dapr.runtime.wfengine")
 	errExecutionAborted = errors.New("execution aborted")
 )
 
+// wfConfig is the configuration for the workflow engine
+type wfConfig struct {
+	AppID             string
+	workflowActorType string
+	activityActorType string
+}
+
+// NewWorkflowConfig creates a new workflow engine configuration
+func NewWorkflowConfig(appID string) *wfConfig {
```

Comment:
why is a pointer to this struct returned? Just a normal struct is enough here?
Am I correct in saying that the value is written once and accessed multiple times, so in that case why is a pointer needed here?

Reply:
Nice point. It doesn't need to be mutable. And, with a pointer, it would also need to do `escape analysis` and it might also land up in heap, where it is not needed actually. I will change this one.